wdl-cli/src/analysis.rs

```diff
@@ -194,11 +210,32 @@ fn get_diagnostics_config(exceptions: &HashSet<String>) -> DiagnosticsConfig {
     }))
 }
 
-/// Gets a lint visitor with the excepted rules removed.
-fn get_lint_visitor(exceptions: &HashSet<String>) -> Linter {
-    Linter::new(wdl_lint::rules().into_iter().filter(|rule| {
-        !exceptions
-            .iter()
-            .any(|exception| exception.eq_ignore_ascii_case(rule.id()))
-    }))
+/// Gets a lint visitor with the rules depending on provided options.
+///
+/// `enabled_lint_tags` controls which rules are considered for being added to
+/// the visitor. `disabled_lint_tags` and `exceptions` act as filters on the set
+/// considerd by `enabled_lint_tags`.
+fn get_lint_visitor(
+    enabled_lint_tags: &TagSet,
+    disabled_lint_tags: &TagSet,
+    exceptions: &HashSet<String>,
+) -> Linter {
+    let mut enabled_rules = vec![];
+    let mut disabled_rules = vec![];
+    let linter = Linter::new(wdl_lint::rules().into_iter().filter(|rule| {
+        let enable = enabled_lint_tags.intersect(rule.tags()).count() > 0
+            && disabled_lint_tags.intersect(rule.tags()).count() == 0
+            && !exceptions
+                .iter()
+                .any(|exception| exception.eq_ignore_ascii_case(rule.id()));
+        if enable {
+            enabled_rules.push(rule.id());
+        } else {
+            disabled_rules.push(rule.id());
+        }
+        enable
+    }));
+    info!("enabled lint rules: {:?}", enabled_rules);
+    info!("disabled lint rules: {:?}", disabled_rules);
```

Comment:
this logs once per analysis thread. @peterhuene is there a better way we can log this only once?

Reply:
I was just about to comment on this.

Probably the easiest way is to move the call of `get_lint_visitor` out of the closure (making it an `Option<Linter>` outside of the closure) which is cloned inside the closure and added to the validator if `Some`.

We'd need `Linter` to implement `Clone`, but there's no reason it couldn't be (should probably be a custom `Clone` impl that clones `rules`, but default initializes `document_exceptions`).